### PR TITLE
fix(#9963):Add logic to handle for fileHandle.Close()

### PR DESCRIPTION
**Pull request by my-git9**
Add logic to handle for fileHandle.Close()

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/cert/cert.go [hunk 1]
```diff
@@ -130,7 +131,11 @@ func ParseTLSCertificatesFromPath(sourceFile string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer fileHandle.Close()
+	defer func() {
+		if err = fileHandle.Close(); err != nil {
+			log.Errorf("error closing file %q: %v", fileHandle.Name(), err)
+		}
```
Comment (crenshaw-dev) [inline util/cert/cert.go:137]: @my-git9 your work inspired some additional logging changes: https://github.com/argoproj/argo-cd/pull/10256/files#diff-25220a29fb80773c52f3ff40158cbfbe138517c2d08b8a204c7091af468b82aeR171

Would you mind adding this to the log statement?

```
.WithFields(log.Fields{
				common.SecurityField:    common.SecurityMedium,
				common.SecurityCWEField: 775,
			})
```

That'll just make it easier for folks to monitor these issues.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
One enhancement request. Thanks as always @my-git9!

File: util/cert/cert.go (hunk 1 above)
Comment (my-git9) [reply util/cert/cert.go:137]: Sure, I will do it later

Comment (my-git9) [reply util/cert/cert.go]: It has been updated. The previous PR was also changed like this, like this: https://github.com/argoproj/argo-cd/pull/10127/files, do I need to submit another PR to adjust it?

Comment (crenshaw-dev) [reply util/cert/cert.go]: Only if you want to! Adding security logging is helpful, but optional.  :-)

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks!

File: util/cert/cert.go (hunk 1 above)
Comment (my-git9) [reply util/cert/cert.go]: OK, I can try it

**Pull request merged**